### PR TITLE
fix: return an error for vault tokens that will expire before the scheduler can run

### PR DESCRIPTION
**Pull request by ddebko**
### Summary:

Currently the Vault credential store API accepts any token TTL, we validate the TTL and store the next renewal time during create and update. We should return an error if that TTL cannot be honored due to the scheduler interval for running jobs.

### Fix:

Now we compare the ttl value to the runJobsInterval value and return an error to the user when we cannot honor the token renewal.

### Error Output:
`scheduler interval must be greater than token ttl. scheduler jobs interval value: 2s`

### CODE REVIEW

File: internal/credential/vault/repository_credential_store.go [hunk 1]
```diff
@@ -116,6 +117,11 @@ func (r *Repository) CreateCredentialStore(ctx context.Context, cs *CredentialSt
 		return nil, err
 	}
 
+	runJobsInterval := r.scheduler.GetRunJobsInterval()
+	if token.expiration <= runJobsInterval {
+		return nil, errors.Wrap(ctx, fmt.Errorf("scheduler interval must be greater than token ttl. scheduler jobs interval value: %s seconds", strconv.FormatFloat(runJobsInterval.Seconds(), 'f', -1, 64)), op)
```
Comment (louisruch) [inline internal/credential/vault/repository_credential_store.go:122]: Do we have other examples in Boundary where we print a duration? Getting it in seconds is probably fine, but we could just print out using golang's default duration print.

**Review by louisruch — COMMENTED** [review]
LGTM, I did have one question about the duration format for the error message

File: internal/credential/vault/repository_credential_store.go (hunk 1 above)
Comment (ddebko) [reply internal/credential/vault/repository_credential_store.go:122]: I found one example where we print the time.Duration variable using the String() func, which I just tested out and it does look nicer.

**Review by louisruch — APPROVED** [review]
LGTM!

**Pull request merged**